alerts/cluster-monitoring-operator/PrometheusOperatorRejectedResources.md

```diff
@@ -0,0 +1,91 @@
+objects.
+
+## Impact
+
+The custom resources that trigger the alert are ignored by Prometheus Operator.
+As a consequence, they will not be part of the final configuration of the
+Prometheus, Alertmanager, or Thanos Ruler components managed by Prometheus Operator.
+
+## Diagnosis
+
+Find the custom resource type and the namespace in the `description` field
+of the `PrometheusOperatorRejectedResources` alert. You can find this information
+by using the OpenShift web console or the command line interface (CLI).
+
+### Diagnose using the Openshift web console
+
+1. Browse to **Observe** -> **Alerting**.
+2. If the **Filter** for **Alert State** is not set to **Firing**,
+ click the **Filter** drop-down list and select **Firing**.
+3. Search for the `PrometheusOperatorRejectedResources` alert.
+4. Click the alert to view its details.
+5. Scroll down and view the **Description** field. The namespace is displayed as
+a link, followed by the resource type.
+
+After you know the namespace and the resource type, you can search for a recently
+modified custom resource of that type in that namespace.
+**Note**: If the namespace is named `openshift-monitoring`,
+this might be a bug in OpenShift. Please file a support case.
+
+Otherwise, the namespace should be `openshift-user-workload-monitoring`.
+
+For more details about the error, view the logs for the Prometheus Operator deployment
+in the `openshift-user-workload-monitoring` namespace.
+```bash
+oc logs deployment/prometheus-operator -c prometheus-operator -n openshift-user-workload-monitoring
+```
+
+The following is a sample error message about a rejected custom resource:
+```log
+level=warn ts=2023-07-03T20:37:20.740723141Z caller=operator.go:1917 component=prometheusoperator msg="skipping servicemonitor" error="it accesses file system via bearer token file which Prometheus specification prohibits" servicemonitor=quarkus-demo/otel-collector namespace=openshift-user-workload-monitoring prometheus=user-workload
+```
+
+### Diagnose using the CLI
+
+* Get the Alertmanager URL and view a list of alerts that have fired:
+```bash
+### Gets alertmanager URL
+$ ALERT_MANAGER=$(oc get route alertmanager-main -n openshift-monitoring -o jsonpath='{@.spec.host}')
+
+### Gets fired alerts
+$ curl -sk -H "Authorization: Bearer $(oc create token prometheus-k8s -n openshift-monitoring)"  https://$ALERT_MANAGER/api/v2/alerts?filter=alertname=PrometheusOperatorRejectedResources
+```
+
+## Mitigation
+
+Fix the misconfigured custom resource and reapply it to the cluster.
+
+The error that causes the alert to fire often occurs because the Operator cannot
+apply a value from certain fields to create a valid configuration.
+The root causes for this alert can vary depending on the specific configuration and
+deployment of Prometheus Operator.
+Possible causes include the following:
+
+- Violation of file system access rules through a bearer token file or
```

Comment:
Can we rephrase it to make it more clear about what isn't allowed? E.g. "A ServiceMonitor or PodMonitor object references a file to use as bearer token or TLS file which isn't allowed for user-defined monitoring. Instead it is necessary to create a secret holding the credential data in the same namespace  as the ServiceMonitor or PodMonitor object and use a secret key reference in the ServiceMonitor or PodMonitor."

Reply:
Slightly edited version of @simonpasquier's suggested wording:

"A violation can occur when a `ServiceMonitor` or `PodMonitor` object references a file to use as a bearer token or references a TLS file. These configurations are not allowed in user-defined monitoring. Instead, in user-defined monitoring, you must create a secret that contains the credential data in the same namespace as the `ServiceMonitor` or `PodMonitor` object and use a secret key reference in the `ServiceMonitor` or `PodMonitor` configuration."

WDYT?